### PR TITLE
Merge contacts frontend into government frontend

### PR DESCRIPTION
**Pull request by mcgoooo**
This is the first part of the template consolidation work. This renders contacts frontend in government frontend. 

part of :- https://trello.com/c/0Ree9gt0/140-3-render-contacts-in-government-frontend-l

it handles all the rendering, and Webchat that was handled by contacts frontend. Please note, the webchat was taken in as is, as it is being [handled in firebreak](https://trello.com/c/BkJnY6tj/3-webchat-improvements) 

The breadcrumbs are awaiting on work from the [finding things team](https://trello.com/c/0OvN0dtQ/412-make-sure-contacts-have-the-correct-parent-set).

the contacts template is using govspeak css and will be having a going over with the content team to make it more consistent. the reason Govspeak was used was to introduce more consistency as opposed to specific styles per template.

you can see changes supplied by wraith here https://template-consolidation-contacts-diff.surge.sh/

this pull is dependent on [updating the schemas](https://github.com/alphagov/govuk-content-schemas/pull/533)

| Before | After |
|--|--|
|multi_phantomjs_local | multi_phantomjs_production |

### CODE REVIEW

**Comment by boffbowsh** [comment]
This didn't previously have the beta banner and I can't easily see a description of why and where it got added.

**Comment by nickcolley** [comment]
We removed the beta banner since no product team is iterating contacts at the moment. So it felt like something that should have been removed a long time ago.

I'd be happy to put it back if there's a good reason to though. 👍

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -1 +1,15 @@
+/* eslint-disable */
 //= require_tree ./modules
+//= require webchat.js
```
Comment (fofr) [inline app/assets/javascripts/application.js]: Could we include this only on pages that have webchat?

**Comment by fofr** [comment]
Can the commit message for 3cc1e1ca33a90243baf82e7890d3e496c410940d be updated with a link to the history of the original webchat feature?

File: app/assets/javascripts/application.js (hunk 1 above)
Comment (nickcolley) [inline app/assets/javascripts/application.js]: I'd be 👍  for that, seems like a good thing.

File: test/integration/contact_test.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+      html = Nokogiri::HTML.parse(content)
+      assert_not_nil html.at_css("h2#phone-title")
+      assert_not_nil html.at("p:contains(\"#{first_phone}\")")
+      assert_not_nil html.at("p:contains(\"24 hours a day, 7 days a week\")")
+    end
+  end
+
+
+  test "post are rendered" do
```
Comment (fofr) [inline test/integration/contact_test.rb]: "post are rendered" (also, whitespace)

File: test/presenters/contact_presenter_test.rb [hunk 3]
```diff
@@ -0,0 +1,81 @@
+require 'presenter_test_helper'
+
+class ContactPresenterTest
+  class PresentedContact < PresenterTestCase
+    def format_name
+      "contact"
+    end
+
+    test 'presents the format' do
+      assert_equal schema_item['format'], presented_item.format
```
Comment (fofr) [inline test/presenters/contact_presenter_test.rb]: I think we can remove this test

File: test/presenters/contact_presenter_test.rb [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    test 'presents online form links' do
+      assert_equal schema_item['details']['contact_form_links'].first['link'], presented_item.online_form_links.first[:url]
+    end
+
+    test 'presents online form body' do
+      assert_equal schema_item['details']['more_info_contact_form'], presented_item.online_form_body
+    end
+
+    test 'phone returns correctly' do
+      assert_equal schema_item['details']['phone_numbers'][0]["number"], presented_item.phone[0][:numbers][0][:number]
```
Comment (fofr) [inline test/presenters/contact_presenter_test.rb]: I'd save `schema_item['details']['phone_numbers'][0]` in a variable

File: test/integration/contact_test.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    setup_and_visit_content_item('contact')
+    within_component_govspeak do |component_args|
+      content = component_args.fetch("content")
+      html = Nokogiri::HTML.parse(content)
+
+      assert_not_nil html.at_css("h2#post-title")
+      assert_not_nil html.at_css(".street-address")
+    end
+  end
+end
```
Comment (fofr) [inline test/integration/contact_test.rb]: There's no test that the related links component is rendered with the correct parameters.

**Comment by fofr** [comment]
What's holding this up?

It also needs a rebase with master after travel advice was added.

**Comment by nickcolley** [comment]
Just me trying to find time to update this PR, I'm close to getting in a few commits that address the rest of your comments. ⏲

**Comment by nickcolley** [comment]
I've updated the PR so that Webchat is only loaded when it's needed. We should test this in production mode that the webchat payload is only loaded when needed etc.

File: app/assets/javascripts/modules/webchat.js [hunk 6]
```diff
@@ -0,0 +1,151 @@
+(function (global) {
```
Comment (fofr) [inline app/assets/javascripts/modules/webchat.js]: This doesn't match the [module pattern](https://github.com/alphagov/govuk_frontend_toolkit/blob/master/docs/javascript.md#modules). Could it live in the parent directory or a different subdir? Then `//= require_tree ./modules` continues to work as before.

eg, it's not in the form:

```js
GOVUK.Modules.TrackShareButtonClicks = function () {
    this.start = function (element) {
```

File: spec/javascripts/helpers/cookie-functions.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+      while (cookie.charAt(0) == ' ') {
+        cookie = cookie.substring(1, cookie.length);
+      }
+      if (cookie.indexOf(nameEQ) === 0) {
+        return decodeURIComponent(cookie.substring(nameEQ.length));
+      }
+    }
+    return null;
+  };
+}).call(this);
```
Comment (fofr) [inline spec/javascripts/helpers/cookie-functions.js]: Do these methods need to work or should they be stubbed/mocked for the tests?

File: spec/javascripts/support/jasmine.yml [hunk 8]
```diff
@@ -17,7 +17,11 @@ css_dir: "app/assets/stylesheets"
 # list of file expressions to include as source files
 # relative path from src_dir
 src_files:
-  - "application.{js.coffee,js,coffee}"
+  - "../../../spec/javascripts/helpers/jquery-1.12.4.js"
+  - "../../../spec/javascripts/helpers/jasmine-jquery-2.0.5.js"
+  - "../../../spec/javascripts/helpers/cookie-functions.js"
```
Comment (fofr) [inline spec/javascripts/support/jasmine.yml]: Where is this used? I see no other references to `cookie` in the spec that's added.

File: spec/javascripts/track-share-button-clicks.spec.js [hunk 9]
```diff
@@ -9,18 +9,21 @@ describe('A share button click tracker', function () {
...
   })
 
   it('tracks click events on share buttons', function () {
     spyOn(GOVUK.analytics, 'trackShare')
-    element = $('<div>\
-      <a href="#" class="js-share-facebook" data-network="facebook">Share</a>\
-      <a href="#" class="js-share-twitter" data-network="twitter">Share</a>\
-    </div>')
+    element = $(
+      '<div>' +
```
Comment (fofr) [inline spec/javascripts/track-share-button-clicks.spec.js]: Why was this updated?

I prefer writing long JS strings with `\` line-breaks, it's similar to the way we write Ruby arrays using `%w{}` – it requires less syntax.

**Comment by fofr** [comment]
Don't forget to add Contacts to the README table: https://github.com/alphagov/government-frontend#formats

File: spec/javascripts/track-share-button-clicks.spec.js (hunk 9 above)
Comment (nickcolley) [inline spec/javascripts/track-share-button-clicks.spec.js]: It makes standard.js sad

Comment (fofr) [inline spec/javascripts/track-share-button-clicks.spec.js]: That makes me sad.

Comment (mcgoooo) [inline spec/javascripts/track-share-button-clicks.spec.js]: es6 template literals?
https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Template_literals

File: README.md [hunk 10]
```diff
@@ -17,6 +17,7 @@ Not all formats that this app can handle are rendered by it in production.
 | Case study | [View on GOV.UK](https://www.gov.uk/government/case-studies/2013-elections-in-swaziland) | Migrated |
 | Coming soon | | Rendered by Whitehall |
 | Consultation | [View on GOV.UK](https://www.gov.uk/government/consultations/soft-drinks-industry-levy) | Migrated |
+| contacts | [View on GOV.UK](https://www.gov.uk/government/organisations/hm-revenue-customs/contact/alcohol-duties-national-registration-unit) | Currently rendered by Contacts frontend |
```
Comment (fofr) [inline README.md]: Nitpick: Should be uppercase

File: spec/javascripts/track-share-button-clicks.spec.js (hunk 9 above)
Comment (nickcolley) [inline spec/javascripts/track-share-button-clicks.spec.js]: We don't transpile JavaScript but if we started that'd be the best choice.

**Pull request merged**